step-capstone/src/components/TripList.js

```diff
@@ -33,30 +38,64 @@ class TripList extends React.Component{
     .catch(error => {
       console.log("Error Getting Trips")
       console.error(error)
-    });
+    })
+  }
+
+  componentDidMount() {
+    this.loadTrips();
+    
   }
 
+
   handleOpenTrip(tripId) {
     this.setState({selectedTrip : tripId})
     sessionStorage.setItem("tripId",tripId);
   }
 
+  handleAddTrip(newTrip) {
+    console.log("Adding new trip to database...");
+    const reference = "/users/" + this.state.userId + "/trips";
+    this.context.addTrip(reference, newTrip);
+    this.loadTrips();
+  }
+
+  handleDeleteTrip(tripId) {
+    const reference = "/users/" + this.state.userId + "/trips/" + tripId;
+    console.log(reference);
+    this.context.deleteTrip(reference).then(() => this.loadTrips());
+  }
+
   render () {
     const trips = this.state.trips;
     return (
       <div>
         {this.state.selectedTrip ? <Redirect to = {"/trips/"+this.state.selectedTrip}/> :
-          <Grid id="trips" container spacing = {2}> {
-            trips.map((trip) => {
-              return (
-                <Grid item>
-                  <TripItemComponent key={trip.id} data={trip.data()} tripId={trip.id} onOpenTrip={this.handleOpenTrip} />
-                </Grid>
-              );
-            })
-          }
+          <Grid id="trips" 
+            container spacing = {2} 
+            style={{position: "absolute", 
+                    top: "100px", 
+                    left: "100px"}}
+          > 
+            {trips.map((trip) => {
+                return (
+                  <Grid item>
+                    <TripItemComponent 
+                      key = {trip.id} 
+                      data = {trip.data()} 
+                      tripId = {trip.id} 
+                      onOpenTrip ={this.handleOpenTrip}
+                      onDeleteTrip = {this.handleDeleteTrip} />
+                  </Grid>
+                );
+              })
+            }
+                  <Grid item>
+                    <AddTrip
+                      onAddTrip = {this.handleAddTrip}/>
+                  </Grid>
```

Comment:
unindennt a couple i think should be inline with trips.map

Reply:
Yes, so I was trying to put AddTrip in the same indentation as TripItemComponent. Do you think it's a bad idea?